controllers/quote.js

```diff
@@ -157,37 +172,103 @@ exports.get = function *() {
         'source transfers')
     }
   } else {
-    sourceExpiryDuration = destinationExpiryDuration +
-      config.expiry.minMessageWindow
+    sourceExpiryDuration = destinationExpiryDuration + config.expiry.minMessageWindow
+  }
+
+  let source_ledger = params.source_ledger
+  if (!source_ledger) {
+    if (params.source_account) source_ledger = yield getAccountLedger(params.source_account)
+    else throw new Error('Missing required parameter: source_ledger or source_account')
   }
 
-  let paymentTemplate = {
+  let destination_ledger = params.destination_ledger
+  if (!destination_ledger) {
+    if (params.destination_account) destination_ledger = yield getAccountLedger(params.destination_account)
+    else throw new Error('Missing required parameter: destination_ledger or destination_account')
+  }
+
+  return {
+    destinationExpiryDuration: destinationExpiryDuration,
+    sourceExpiryDuration: sourceExpiryDuration,
+    source_amount: params.source_amount,
+    destination_amount: params.destination_amount,
+    source_ledger: source_ledger,
+    destination_ledger: destination_ledger,
+    source_account: params.source_account || null,
+    destination_account: params.destination_account || null
+  }
+}
+
+function makeQuoteArgs (query) {
+  return {
+    source_ledger: query.source_ledger,
+    destination_ledger: query.destination_ledger,
+    source_amount: query.source_amount,
+    destination_amount: query.destination_amount
+  }
+}
+
+function makePaymentTemplate (query, quote) {
+  const source_amount = quote.source_amount.toFixed(2, 2)
```

Comment:
I'm not sure about the two decimal places, that's always been there. As for the [rounding modes](https://github.com/MikeMcl/bignumber.js/blob/ad704dc44f9e2de5cd8bf027b0ff3fcc18e83811/bignumber.js#L63-L75), the source amount needs to be rounded up (`CEIL`) -- if it was rounded down, it would cause the transaction to be created at an unacceptable rate, and it would immediately be rejected.